pkg/plugin/trivy/plugin.go

```diff
@@ -1144,8 +1150,13 @@ func (p *plugin) getPodSpecForStandaloneFSMode(ctx starboard.PluginContext, conf
 		InitContainers:               []corev1.Container{initContainerCopyBinary, initContainerDB},
 		Containers:                   containers,
 		SecurityContext:              &corev1.PodSecurityContext{},
-		NodeName:                     spec.NodeName,
-	}, secrets, nil
+	}
+
+	if !ctx.GetStarboardConfig().IsScanJobsInSameNamespace() {
+		podSpec.NodeName = spec.NodeName
```

Comment:
I see that we take the value form spec.NodeName based on p.objectResolver.GetNodeName(). But why don't we use objectResolver directly here?

Reply:
GetNodeName() returns error as well and when there is error then we dont want to proceed for building other items(initcontainer, vol, etc). So get nodeName at the start and using it here. wdyt

Reply:
We can call it as early as possible, but my point is that adding a dedicated nodeName variable is more readable then updating spec.

Reply:
ohh, I got the point. thanks.